src/WithdrawalQueueManager.sol

```diff
@@ -0,0 +1,353 @@
+            revert WithdrawalAlreadyProcessed(tokenId);
+        }
+
+        if (!isFinalized(request)) {
+            revert NotFinalized(block.timestamp, request.creationTimestamp, secondsToFinalization);
+        }
+
+        withdrawalRequests[tokenId].processed = true;
+        uint256 unitOfAccountAmount = calculateRedemptionAmount(request.amount, request.redemptionRateAtRequestTime);
+        pendingRequestedRedemptionAmount -= unitOfAccountAmount;
+
+        _burn(tokenId);
+        redeemableAsset.burn(request.amount);
+
+
+        uint256 feeAmount = calculateFee(unitOfAccountAmount, request.feeAtRequestTime);
+        uint256 netUnitOfAccountAmount = unitOfAccountAmount - feeAmount;
+
+        uint256 currentBalance = redemptionAssetsVault.availableRedemptionAssets();
+        if (currentBalance < unitOfAccountAmount) {
+            revert InsufficientBalance(currentBalance, unitOfAccountAmount);
+        }
+
+        redemptionAssetsVault.transferRedemptionAssets(receiver, netUnitOfAccountAmount);
+        
+        if (feeAmount > 0) {
+            redemptionAssetsVault.transferRedemptionAssets(feeReceiver, feeAmount);
+        }
+
+        emit WithdrawalClaimed(tokenId, msg.sender, receiver, request);
+    }
+
+    /**
+     * @notice Allows a batch of withdrawals to be claimed by their respective token IDs.
+     * @param tokenIds An array of token IDs representing the withdrawal requests to be claimed.
+     */
+    function claimWithdrawals(uint256[] calldata tokenIds, address[] calldata receivers) external {
+
+        if (tokenIds.length != receivers.length) {
+            revert ArrayLengthMismatch(tokenIds.length, receivers.length);
+        }
+        for (uint256 i = 0; i < tokenIds.length; i++) {
+            claimWithdrawal(tokenIds[i], receivers[i]);
+        }
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  ADMIN  -------------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    function setSecondsToFinalization(uint256 _secondsToFinalization) external onlyRole(WITHDRAWAL_QUEUE_ADMIN_ROLE) {
+        if (_secondsToFinalization > MAX_SECONDS_TO_FINALIZATION) {
+            revert SecondsToFinalizationExceedsLimit(_secondsToFinalization);
+        }
+        emit SecondsToFinalizationUpdated(secondsToFinalization, _secondsToFinalization);
+        secondsToFinalization = _secondsToFinalization;
+    }
+
+    /// @notice Sets the withdrawal fee percentage.
+    /// @param feePercentage The fee percentage in basis points.
+    function setWithdrawalFee(uint256 feePercentage) external onlyRole(WITHDRAWAL_QUEUE_ADMIN_ROLE) {
+        if (feePercentage > FEE_PRECISION) {
+            revert FeePercentageExceedsLimit();
+        }
+        withdrawalFee = feePercentage;
+        emit WithdrawalFeeUpdated(feePercentage);
```

Comment:
just like the above and below functions include the old and new value in the event args.